### PR TITLE
Set path filters in favor of the required-job action

### LINKED ISSUE #14507: CI: explore trigger required checks on special comment
## 🚀 Feature

How about setting that the check is run on specific comments, so the scenario would be:
- if you edit a particular codebase, your check is started automatically because of the relevance
- if your change is small, and some check was not triggered just before the merge, you may request to run these checks to satisfy and pass them as required jobs

see inspiration: https://github.com/orgs/community/discussions/25389#discussioncomment-3247738

### Motivation

We would remove most of our skip hacks and kee straight reliable CI

### Alternatives

find some similar solution for grouping as we had with 
- #14165
- #14170
- #14363, #14294
- #14456

cc @carmocca @akihironitta @borda

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

I added the [check-group](https://github.com/tianhaoz95/check-group) code into our Probot in https://github.com/carmocca/probot in https://github.com/Lightning-AI/lightning/pull/14621

Before this is merged, we need to **unmark all jobs as required but check-group, WIP, pre-commit, check-schema**

Closes #14507
Closes #14565 

### Does your PR introduce any breaking changes? If yes, please list them.

None

### CODE REVIEW

File: .github/workflows/probot-check-group.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Probot
+
+on:
+  check_run:
```
Comment (carmocca) [inline .github/workflows/probot-check-group.yml:4]: I can't get this event to work... It's blocked on it

**Comment by carmocca** [comment]
@probot pls

File: .github/workflows/probot-check-group.yml (hunk 1 above)
Comment (carmocca) [reply .github/workflows/probot-check-group.yml]: Okay. I'm stupid. The docs say:

> Note: This event will only trigger a workflow run if the workflow file is on the default branch.

File: .azure/gpu-benchmark.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ trigger:
       - "master"
       - "release/*"
       - "refs/tags/*"
+  # TODO: add paths
```
Comment (carmocca) [inline .azure/gpu-benchmark.yml:15]:
```suggestion
```

**Comment by carmocca** [comment]
@probot pls

**Comment by carmocca** [comment]
@probot pls

File: .azure/gpu-tests.yml [hunk 3]
```diff
@@ -12,15 +12,26 @@ trigger:
...
+      - ".azure/gpu-tests.yml"
+      - "examples/run_ddp_examples.sh"
+      - "examples/convert_from_pt_to_pl/**"
+      - "examples/run_pl_examples.sh"
+      - "examples/pl_basics/backbone_image_classifier.py"
+      - "examples/pl_basics/autoencoder.py"
+      - "examples/pl_loops/mnist_lite.py"
+      - "examples/pl_fault_tolerant/automatic.py"
+      - "examples/test_pl_examples.py"
+      - "examples/pl_integrations/dali_image_classifier.py"
```
Comment (Borda) [inline .azure/gpu-tests.yml:26]:
```suggestion
      - "examples/pl_*/**"
      - "examples/test_pl_*.py"
```

File: .github/workflows/probot-check-group.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+on:
+  pull_request: {}
+
+jobs:
+  required-jobs:
+    runs-on: ubuntu-latest
+    # if this timeout triggers, then the job needs to be manually restarted through the GitHub interface
+    timeout-minutes: 60
+    steps:
+      - uses: carmocca/probot@v2
```
Comment (Borda) [inline .github/workflows/probot-check-group.yml:12]: What are the v1 & v2? Can we update all to v2?

File: .azure/gpu-tests.yml (hunk 3 above)
Comment (carmocca) [reply .azure/gpu-tests.yml:26]: Note all pl examples are checked in our CI, so we don't want to do this

File: .github/workflows/probot-check-group.yml (hunk 4 above)
Comment (carmocca) [reply .github/workflows/probot-check-group.yml:12]: v1 has the cc-bot
v2 the check-group integration (WIP)

There's no need to upgrade if v1 works for cc-bot.

File: .azure/ipu-tests.yml [hunk 5]
```diff
@@ -7,6 +7,12 @@ trigger:
...
       - release/*
       - refs/tags/*
+  paths:
+    include:
+      - ".azure/ipu-tests.yml"
+      - "requirements/pytorch/**"
+      - "src/pytorch_lightning/**"
+      - "tests/tests_pytorch/**"
 
 pr:
```
Comment (carmocca) [inline .azure/ipu-tests.yml:17]: FIXME

File: .azure/gpu-tests.yml [hunk 6]
```diff
@@ -12,15 +12,26 @@ trigger:
...
+      - "requirements/pytorch/**"
+      - "src/pytorch_lightning/**"
+      - "tests/tests_pytorch/**"
 
 pr:
```
Comment (carmocca) [inline .azure/gpu-tests.yml:31]: FIXME

File: .github/workflows/probot-check-group.yml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+name: Probot
+
+on:
+  pull_request: {}
```
Comment (carmocca) [inline .github/workflows/probot-check-group.yml:4]:
```suggestion
  pull_request_target:
```

File: .azure/gpu-tests.yml [hunk 8]
```diff
@@ -12,14 +12,42 @@ trigger:
       - "master"
       - "release/*"
       - "refs/tags/*"
+  paths:
+    include:
```
Comment (carmocca) [inline .azure/gpu-tests.yml:16]: FIXME: add Lite

File: .azure/ipu-tests.yml [hunk 9]
```diff
@@ -7,10 +7,24 @@ trigger:
       - master
       - release/*
       - refs/tags/*
+  paths:
+    include:
```
Comment (carmocca) [inline .azure/ipu-tests.yml:11]: FIXME: add Lite

File: .github/workflows/ci-pytorch-test-conda.yml [hunk 10]
```diff
@@ -6,6 +6,12 @@ on:
     branches: [master, "release/*"]
   pull_request:
     branches: [master, "release/*"]
+    paths:
```
Comment (carmocca) [inline .github/workflows/ci-pytorch-test-conda.yml:9]: FIXME: add Lite

File: .github/workflows/ci-pytorch-test-full.yml [hunk 11]
```diff
@@ -7,6 +7,12 @@ on:
   pull_request:
     branches: [master, "release/*"]
     types: [opened, reopened, ready_for_review, synchronize]  # add `ready_for_review` since draft is skipped
+    paths:
```
Comment (carmocca) [inline .github/workflows/ci-pytorch-test-full.yml:10]: FIXME: add Lite

File: .github/workflows/ci-pytorch-test-full.yml [hunk 12]
```diff
@@ -7,6 +7,15 @@ on:
...
+      - "requirements/pytorch/**"
+      - "src/pytorch_lightning/**"
+      - "tests/tests_pytorch/**"
+      - "setup.cfg"  # includes pytest config
+      - ".github/workflows/ci-pytorch-test-full.yml"
+      - "requirements/lite/**"
+      - "src/lightning_lite/**"
+      - "tests/tests_lite/**"
```
Comment (awaelchli) [inline .github/workflows/ci-pytorch-test-full.yml:18]: why is it included here? Changes to pytorch won't affect lite testing at all. 
We have ci-lite-full

File: .azure/gpu-tests.yml [hunk 13]
```diff
@@ -12,14 +12,50 @@ trigger:
...
+      - "examples/pl_fault_tolerant/automatic.py"
+      - "examples/test_pl_examples.py"
+      - "examples/pl_integrations/dali_image_classifier.py"
+      - "requirements/pytorch/**"
+      - "src/pytorch_lightning/**"
+      - "tests/tests_pytorch/**"
+      - "setup.cfg"  # includes pytest config
+      - "requirements/lite/**"
+      - "src/lightning_lite/**"
+      - "tests/tests_lite/**"
```
Comment (awaelchli) [inline .azure/gpu-tests.yml:33]: same question here

File: .github/workflows/ci-pytorch-test-full.yml (hunk 12 above)
Comment (carmocca) [reply .github/workflows/ci-pytorch-test-full.yml:18]: In the checkgroup config, it is set that changes to Lite should require that PL tests pass

Which means that a change to Lite needs to launch the PL tests. This is what this configures

File: .azure/gpu-tests.yml (hunk 13 above)
Comment (carmocca) [reply .azure/gpu-tests.yml:33]: Samw answer

File: .github/workflows/ci-pytorch-test-full.yml (hunk 12 above)
Comment (carmocca) [reply .github/workflows/ci-pytorch-test-full.yml:18]: https://github.com/Lightning-AI/lightning/blob/326e5afafd35ab89545a739a347a5d8db27cec81/.github/checkgroup.yml#L169-L178

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**